test/concerns/counter_cache_test.rb

```diff
@@ -0,0 +1,65 @@
+require_relative '../environment'
+
+class CounterCacheTest < ActiveSupport::TestCase
+  def test_counter_cache_when_creating
+    AncestryTestDatabase.with_model :depth => 2, :width => 2, :counter_cache => true do |model, roots|
+      roots.each do |lvl0_node, lvl0_children|
+        assert_equal 2, lvl0_node.reload.children_count
+      end
+    end
+  end
+
+  def test_counter_cache_when_destroying
+    AncestryTestDatabase.with_model :depth => 2, :width => 2, :counter_cache => true do |model, roots|
+      parent = roots.first.first
+      child = parent.children.first
+      assert_difference 'parent.reload.children_count', -1 do
+        child.destroy
+      end
+    end
+  end
+
+  def test_counter_cache_when_updating_parent
+    AncestryTestDatabase.with_model :depth => 2, :width => 2, :counter_cache => true do |model, roots|
+      parent1 = roots.first.first
+      parent2 = roots.last.first
+      child = parent1.children.first
+
+      assert_difference 'parent1.reload.children_count', -1 do
+        assert_difference 'parent2.reload.children_count', 1 do
+          child.update parent: parent2
```

Comment:
update_attributes ?

Reply:
I'm still not happy about using `update` here

Reply:
but `update_attributes` is the alias of `update`